src/main/java/com/googlecode/protobuf/format/JsonFormat.java

```diff
@@ -875,6 +885,36 @@ protected void mergeField(Tokenizer tokenizer,
             mergeField(tokenizer, extensionRegistry, builder);
         }
     }
+    
+    private void handleUnknownField(Tokenizer tokenizer,    										
+    		Message.Builder builder,
+    		int fieldNumber) throws ParseException {    	
+    	tokenizer.tryConsume(":");
+    	if ("{".equals(tokenizer.currentToken())) {
+    		// Message structure
+    		tokenizer.consume("{");
+    		do {
+    			tokenizer.consumeIdentifier();
+    			handleUnknownField(tokenizer, builder, fieldNumber);
+    		} while (tokenizer.tryConsume(","));

```

Comment:
What happens if there's an object with trailing comma?  Do we not care about that case, since this package shouldn't be generating that output?

```
{
  "foo": "bar",
  "baz": "buz",
}
```